static/src/javascripts/projects/common/modules/commercial/dfp/get-adverts.js

```diff
@@ -0,0 +1,18 @@
+define([
+    'common/modules/commercial/dfp/private/dfp-env',
+    'common/modules/commercial/dfp/private/get-advert-by-id',
+    'common/modules/commercial/dfp/dfp-obj'
+], function (dfpEnv, getAdvertById, dfp) {
+    dfp.getAdverts = getAdverts;
+    return getAdverts;
+
+    function getAdverts(isWithAllAds) {
```

Comment:
Good point. `adverts` may contain ads that are empty, i.e. when DFP returned no creative. They used to be removed from the array, but instead I keep them so we can track the lifecycle of all ad slots, no matter what. Indeed, the parameter name is misleading.